test/kubemark/start-kubemark.sh

```diff
@@ -446,55 +450,92 @@ function setup_proxy {
   export KUBERNETES_SCALEOUT_PROXY=false
 }
 
-detect-project &> /dev/null
+function create_TP() {
+  local id=${1:-0}
+  echo "DBG: creating TP ${id}"
+  export TENANT_PARTITION_SEQUENCE=${id}
+  export KUBEMARK_CLUSTER_KUBECONFIG="${TP_KUBECONFIG}-${id}"
+  create-kubemark-master
+  echo "DBG: TP ${id} created"
+}
 
-rm /tmp/saved_tenant_ips.txt >/dev/null 2>&1 || true
+function create_RP() {
+  local id=${1:-0}
+  echo "DBG: creating RP ${id}"
+  export RESOURCE_PARTITION_SEQUENCE=${id}
+  export KUBEMARK_CLUSTER_KUBECONFIG="${RP_KUBECONFIG}-${id}"
+  create-kubemark-master
+  echo "DBG: RP ${id} created"
+}
+
+detect-project &> /dev/null
 
 ### master_metadata is used in the cloud-int script to create the GCE VMs
 #
 MASTER_METADATA=""
 
+### to upload etcd image / binary tar once
```

Comment:
the test_resource_upload() can be added to the very beginning of the test setup and ensure it is done. so that this check can be removed

Reply:
admin cluster also needs this uploading, and it is directly set up by kube-up.sh script; starting partition clusters also eventually call into kube-up.sh; if we don't want to change this structure, we have to have some sort of condition check in place.